core/lib/types/src/snapshots.rs

```diff
@@ -0,0 +1,80 @@
+use crate::commitment::L1BatchWithMetadata;
+use crate::{StorageKey, StorageValue, H256};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use zksync_basic_types::{L1BatchNumber, MiniblockNumber};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AllSnapshots {
+    pub snapshots: Vec<SnapshotMetadata>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SnapshotMetadata {
+    pub l1_batch_number: L1BatchNumber,
+    pub generated_at: DateTime<Utc>,
+    pub factory_deps_filepath: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SnapshotHeader {
+    pub l1_batch_number: L1BatchNumber,
```

Comment:
Nit: This is duplicated in `last_l1_batch_with_metadata.header.number`.

Reply:
I know, but I feel that it's very natural that snapshot header contains l1_batch_number that is it's primary_key, that header is just an embedded object related to the snapshot header that accidentally happens to have this field